src/lumigo_opentelemetry/libs/sampling.py

```diff
@@ -0,0 +1,118 @@
+import os
+import re
+from typing import Optional, Sequence
+
+from opentelemetry.context import Context
+from opentelemetry.sdk.trace.sampling import (
+    Decision,
+    ParentBased,
+    Sampler,
+    SamplingResult,
+)
+from opentelemetry.trace import Link, SpanKind, get_current_span
+from opentelemetry.trace.span import TraceState
+from opentelemetry.util.types import Attributes
+
+from lumigo_opentelemetry.libs.environment_variables import (
+    AUTO_FILTER_HTTP_ENDPOINTS_REGEX,
+)
+
+
+class AttributeSampler(Sampler):
+    """Sampler that returns a decision based on the provided attributes.
+
+    Sampling takes place before request / response hooks are called, so our custom attributes are
+    not yet available.
+
+    By default, the decision is RECORD_AND_SAMPLE.
+
+    If the AUTO_FILTER_HTTP_ENDPOINTS_REGEX environment variable is set, then the decision is based
+    on whether the request URL matches the given regex. If it matches, then the decision is DROP.
+    Otherwise, the decision is based on the parent span's decision.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def should_sample(
+        self,
+        parent_context: Optional["Context"],
+        trace_id: int,
+        name: str,
+        kind: SpanKind = None,
+        attributes: Attributes = None,
+        links: Optional[Sequence["Link"]] = None,
+        trace_state: "TraceState" = None,
+    ) -> "SamplingResult":
+        decision = Decision.RECORD_AND_SAMPLE
+        attributes_url = _extract_url(attributes)
+        if attributes_url and _should_skip_span_on_route_match(attributes_url):
+            decision = Decision.DROP
+            attributes = None
+
+        return SamplingResult(
+            decision,
+            attributes,
+            _get_parent_trace_state(parent_context),
+        )
+
+    def get_description(self) -> str:
+        return "SkipSampler"
+
+
+LUMIGO_SAMPLER = ParentBased(AttributeSampler())
+
+
+def _extract_url(attributes: Attributes) -> Optional[str]:
```

Comment:
IMO this is too complex, and maybe we should work only with host (if it answer the user's request)

Reply:
the full url (not including querystring) was specified in the requirement...

Reply:
i've added appropriate unit tests, i'm confident this is good